### DISCUSSION TITLE
FreeBSD build target

### DISCUSSION THREAD

**Discussion by mco-system** [dbeaver/dbeaver#19482] [Q&A] [answered]
Hello,

I'm using DBeaver for years on Linux systems and I have now moved to FreeBSD.

I may have missed something but I found no mention of FreeBSD in the repository and no discussion labeled "freebsd".

I naively thought that I could build a FreeBSD version of DBeaver using the "mvn package" command but it only built the Linux, MacOS and Windows versions.

What would it take to have a build target for FreeBSD ?
Is it even possible ? If yes, what is missing to make it ?

Thank for your answers.

**Comment by serge-rider** [comment, marked as answer]
Native FreeBSD platform support is not supported by Eclipse platform.
However you could try FreeBSD port of Eclipse - https://www.freshports.org/java/eclipse and then install DBeaver plugin https://dbeaver.io/update/latest/

**Reply by mco-system** [reply]
Thank you for your reply.

After opening this discussion I tried to compile DBeaver for FreeBSD.
After patching some OS detection code at various place I almost succeeded. 
The last problem resided at the packaging level at the end of the compilation process.  I was not able to solve it.
As you mentioned Eclipse does not natively support FreeBSD and  the FreeBSD version of it is patched to support FreeBSD, so I gave up there.

I never succeeded in installing/using the plugin for Eclipse. Documentation is not enough clear to me to understand how to install/enable and use a plugin in Eclipse (Never used Eclipse before). The plugin seems to be installed  (not even sure about that point !) but I can't see how to use it .
Basically I hoped to have some sort of connection wizard to ask me the host, username, database type, etc)  ans some menu to install some database drivers but I couldn't find anything like this.
I should give it another try later...

My usage is very basic, I use DBeaver as a database browser and client for very basic tasks.

Any help would be appreciated.

**Reply by barulheira** [reply]
In Eclipse, open the DBeaver perspective (usually in the upper right corner).
Here it failed, because it does not detect the FreeBSD platform. Maybe if you patch it, you can succeed.
https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/utils/RuntimeUtils.java
